app/src/main/java/de/mytfg/apps/mytfg/firebase/FbVplan.java

```diff
@@ -50,12 +54,50 @@ void handle(Map<String, String> data) {
         String userclass = api.getUser().getGrade();
         int userrights = api.getUser().getRights();
         String text = "";
+
+        // Calculate ID based on date and day
+        String toHash = MyTFGApi.tsToString(System.currentTimeMillis() / 1000) + day;
+
+        int nextId = -1;
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            // New devices: Update existing notification
+            NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+            nextId = toHash.hashCode();
+            for (StatusBarNotification notification : notificationManager.getActiveNotifications()) {
+                if (notification.getId() == nextId) {
+                    // Last notification still active
+                    ArrayList<String> oldClasses = notification.getNotification().extras.getStringArrayList("classes");
+                    ArrayList<String> oldTeachers = notification.getNotification().extras.getStringArrayList("teachers");
+                    if (oldClasses != null) {
+                        for (String cls : oldClasses) {
+                            if (!classes.contains(cls)) {
+                                classes.add(cls);
+                            }
+                        }
+                    }
+                    if (oldTeachers != null) {
+                        for (String teacher : oldTeachers) {
+                            if (!teachers.contains(teacher)) {
+                                teachers.add(teacher);
+                            }
+                        }
+                    }
```

Comment:
Naja, es kann ja sein, dass man sich in der Schule den Eintrag angesehen hat "1. frei". 
Dann kriegt man auch die Benachrichtigung, wenn der Eintrag wieder gelöscht wird. Wenn das zum "Konflikt" zwischen zwei Benachrichtigungen führt, dann hat man halt eine unnötige Benachrichtigung. Dafür müsste der aber dann doch nur ein paar Minuten "krank" sein und sich recht spontan wieder umentscheiden 😄